compass/logging.py

```diff
@@ -0,0 +1,40 @@
+import inspect
+import functools
+
+
+def log_method_call(method, logger):
+    """
+    Log the module path and file path of a call to a method, e.g.::
+
+      compass calling: compass.landice.tests.dome.decomposition_test.DecompositionTest.run()
+        in /turquoise/usr/projects/climate/mhoffman/mpas/compass/compass/landice/tests/dome/decomposition_test/__init__.py
+
+    Parameters
+    ----------
+    method : method
+        The method of a class that will be run immediately following this call
+
+    logger: logging.Logger
+        The logger to log the method path and file path to
+    """
+    if not inspect.ismethod(method):
+        raise ValueError('The "method" argument must be a method')
+
+    method_name = method.__name__
+    child_class = method.__self__.__class__
+    child_location = f'{child_class.__module__}.{child_class.__name__}'
+    actual_class = None
+    for cls in inspect.getmro(child_class):
```

Comment:
Can you add a brief comment explaining this loop?

Reply:
@trhille, let me know if the added comments work for you.